snuba/query/mql/parser.py

```diff
@@ -996,10 +997,36 @@ def populate_query_from_mql_context(
     return query, mql_context
 
 
+def quantiles_to_quantile(
+    query: Union[CompositeQuery[QueryEntity], LogicalQuery]
+) -> None:
+    """
+    Changes quantiles(0.5)(...) to arrayElement(quantiles(0.5)(...), 1). This is to simplify
+    the API (so that the arrays don't need to be unwrapped) and also avoids bugs where comparing
+    arrays of values to values cause typing errors (e.g. [1] / 1).
+    """
+
+    def transform(exp: Expression) -> Expression:
+        if isinstance(exp, CurriedFunctionCall):
+            if exp.internal_function.function_name in ("quantiles", "quantilesIf"):
```

Comment:
Maybe we would like to expose the function as `quantile` and `quantileIf` to avoid breaking existing implementations that use `arrayElement(quantilesIf(...), 1)`.